controllers/core/openstackclient_controller.go

```diff
@@ -0,0 +1,167 @@
+}
+
+// GetLogger -
+func (r *OpenStackClientReconciler) GetLogger() logr.Logger {
+	return r.Log
+}
+
+// GetScheme -
+func (r *OpenStackClientReconciler) GetScheme() *runtime.Scheme {
+	return r.Scheme
+}
+
+// OpenStackClientReconciler reconciles a OpenStackClient object
+type OpenStackClientReconciler struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	Kclient kubernetes.Interface
+	Log     logr.Logger
+}
+
+//+kubebuilder:rbac:groups=core.openstack.org,resources=openstackclients,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core.openstack.org,resources=openstackclients/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core.openstack.org,resources=openstackclients/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete;
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;
+
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
+func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	_ = context.Background()
+	_ = r.Log.WithValues("openstackclient", req.NamespacedName)
+
+	instance := &corev1beta1.OpenStackClient{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	r.Log.Info("OpenStackClient CR values", "Name", instance.Name, "Namespace", instance.Namespace, "Secret", instance.Spec.OpenStackConfigSecret, "Spec", fmt.Sprintf("%T", instance.Spec), "Image", instance.Spec.ContainerImage)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	h, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	//
+	// create cm holding deployment script and render deployment script.
+	//
+	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel("openstackclient"), map[string]string{})
+	envVars := make(map[string]env.Setter)
+
+	cms := []util.Template{
+		// ScriptsConfigMap
+		{
+			Name:               "openstackclient-sh",
+			Namespace:          instance.Namespace,
+			Type:               util.TemplateTypeScripts,
+			InstanceType:       instance.Kind,
+			AdditionalTemplate: map[string]string{},
+			Labels:             cmLabels,
+		},
+	}
+	err = configmap.EnsureConfigMaps(ctx, h, instance, cms, &envVars)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	clientLabels := map[string]string{
+		"app": "openstackclient",
+	}
+	// Define a new Deployment object
+	depl := deployment.NewDeployment(openstack.ClientDeployment(instance, clientLabels), 5)
```

Comment:
With a deployment type you don't get a consistent/easy consumable name for the pod, which I think is not really user friendly. Also do we need/want this to be scale-able using a deployment type? Wouldn't it be better , like we did in ospdo, to just create a pod which has the name of the CR?

Reply:
Switched it back to a pod. Thanks.